auth.php

```diff
@@ -323,10 +338,88 @@ protected function is_ip_restriction_enabled() {
      * @return object User object.
      */
     protected function create_user(array $data) {
-        // TODO:
-        // 1. Validate user
-        // 2. Create user.
-        // 3. Throw exception if something went wrong.
+        global $DB, $CFG;
```

Comment:
Have you looked at  core_user_external::create_users instead of the custom validation and creation of the user? Any pros and cons?